File: mmupnp/src/main/java/net/mm2d/upnp/empty/EmptyAction.java
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.empty;

import net.mm2d.upnp.Action;
import net.mm2d.upnp.Argument;
import net.mm2d.upnp.ControlPoints;
import net.mm2d.upnp.Service;

import java.io.IOException;
import java.util.Collections;
import java.util.List;
import java.util.Map;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

public class EmptyAction implements Action {
    @Nonnull
    @Override
    public Service getService() {
        return ControlPoints.emptyService();
    }

    @Nonnull
    @Override
    public String getName() {
        return "";
    }

    @Nonnull
    @Override
    public List<Argument> getArgumentList() {
        return Collections.emptyList();
    }

    @Nullable
    @Override
    public Argument findArgument(@Nonnull final String name) {
        return null;
    }

    @Nonnull
    @Override
    public Map<String, String> invokeSync(@Nonnull final Map<String, String> argumentValues) throws IOException {
        throw new IOException("empty object");
    }

    @Nonnull
    @Override
    public Map<String, String> invokeSync(
            @Nonnull final Map<String, String> argumentValues,
            final boolean returnErrorResponse) throws IOException {
        throw new IOException("empty object");
    }

    @Nonnull
    @Override
    public Map<String, String> invokeCustomSync(
            @Nonnull final Map<String, String> argumentValues,
            @Nullable final Map<String, String> customNamespace,
            @Nonnull final Map<String, String> customArguments) throws IOException {
        throw new IOException("empty object");
    }

    @Nonnull
    @Override
    public Map<String, String> invokeCustomSync(
            @Nonnull final Map<String, String> argumentValues,
            @Nullable final Map<String, String> customNamespace,
            @Nonnull final Map<String, String> customArguments,
            final boolean returnErrorResponse) throws IOException {
        throw new IOException("empty object");
    }
}

File: mmupnp/src/main/java/net/mm2d/upnp/empty/EmptyAction.kt
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.empty

import net.mm2d.upnp.Action
import net.mm2d.upnp.Argument
import net.mm2d.upnp.ControlPoints
import net.mm2d.upnp.Service
import java.io.IOException

/**
 * Empty implementation of [Action].
 */
object EmptyAction : Action {
    override val service: Service = ControlPoints.emptyService()
    override val name: String = ""
    override val argumentList: List<Argument> = emptyList()
    override fun findArgument(name: String): Argument? = null

    override suspend fun invoke(
        argumentValues: Map<String, String?>,
        returnErrorResponse: Boolean
    ): Map<String, String> {
        throw IOException("empty object")
    }

    override suspend fun invokeCustom(
        argumentValues: Map<String, String?>,
        customNamespace: Map<String, String>,
        customArguments: Map<String, String>,
        returnErrorResponse: Boolean
    ): Map<String, String> {
        throw IOException("empty object")
    }
}
